src/routes/posts.ts

```diff
@@ -0,0 +1,125 @@
+import * as uuid from 'uuid';
+import * as Joi from '@hapi/joi';
+import { ContainerTypes, ValidatedRequest } from 'express-joi-validation';
+import * as pgsql from '../utils/pgsql';
+import { Post } from '../entities/post';
+import { POSTS_TABLE, USER_HEADER } from '../constants';
+import { BaseRequestSchema } from '../utils/schema';
+import { buildServiceError } from '../utils/error-builder';
+import { Errors } from '../errors';
+
+export const createPostSchema = {
+	body: Joi.object({
+		title: Joi.string().required(),
+	}),
+};
+
+export interface CreatePostRequestSchema extends BaseRequestSchema {
+	[ContainerTypes.Body]: {
+		title: string;
+	};
+}
+
+export async function create(req: ValidatedRequest<CreatePostRequestSchema>) {
+	const newPost = {
+		id: uuid.v4(),
+		user: req.headers[USER_HEADER],
+		createdAt: new Date(),
+		title: req.body.title,
+		votes: 0,
+		upvotePercentage: 0,
+		downvotePercentage: 0,
+	};
+
+	await pgsql.tryCatch(undefined, async (conn) => {
+		await conn.table(POSTS_TABLE).insert(transform(newPost));
+	});
+
+	return newPost;
+}
+
+export async function getAll() {
+	const posts = await pgsql.tryCatch(undefined, async (conn) => {
+		const query = conn
+			.table(POSTS_TABLE)
+			.select('*')
+			.orderBy([
```

Comment:
Imagine that you have millions of records and requests for this endpoint. Do you think this solution would have the desired performance?

Reply:
There are definitely some improvements needed for that sort of load. For starters I would add indexes to the columns used for sorting in order to improve query performance. 
Something else that would definitely help in a real world scenario would be to implement pagination on the endpoint, given that most users would likely never need more than the first X posts.
Since there are no differences to what users see I would also add a short lived cache to reduce the number of DB queries even further, speeding up response times.